<img src="cover_page.jpg" alt="Alt text" width="700"/>

In [44]:
import pandas as pd
import numpy as np

In [45]:
# Leemos dataset
data = pd.read_csv("Sales_200_days.csv")


In [46]:
# Explorar el dataset
data.head()

TV  Radio  Newspaper  Sales
0  230.1   37.8       69.2   22.1
1   44.5   39.3       45.1   10.4
2   17.2   45.9       69.3    9.3
3  151.5   41.3       58.5   18.5
4  180.8   10.8       58.4   12.9

#**Pearson's Correlation Coefficient**

**Combination: Newspaper-Sales**

In [47]:
# Calculate Pearson's correlation coefficient
correlation_matrix = np.corrcoef(data["Newspaper"], data["Sales"])

correlation_matrix

array([[1.        , 0.22829903],
       [0.22829903, 1.        ]])

In [48]:
# The correlation coefficient between x and y is located at position (0, 1) or (1, 0) in the matrix
correlation_coefficient = correlation_matrix[0, 1]

#Create new column to assign the corelation value
data["Newspaper-Sales"] = correlation_coefficient

print("Pearson's correlation coefficient for the inputs Newspaper-Sales:", correlation_coefficient)

Pearson's correlation coefficient for the inputs Newspaper-Sales: 0.22829902637616528


**Combination: Radio-Sales**

In [49]:
# Calculate Pearson's correlation coefficient
correlation_matrix = np.corrcoef(data["Radio"], data["Sales"])

correlation_matrix

array([[1.        , 0.57622257],
       [0.57622257, 1.        ]])

In [50]:
# The correlation coefficient between x and y is located at position (0, 1) or (1, 0) in the matrix
correlation_coefficient = correlation_matrix[0, 1]

#Create new column to assign the corelation value
data["Radio-Sales"] = correlation_coefficient



print("Pearson's correlation coefficient for the inputs Radio-Sales:", correlation_coefficient)

Pearson's correlation coefficient for the inputs Radio-Sales: 0.5762225745710551


**Combination: TV-Sales**

In [51]:
# Calculate Pearson's correlation coefficient
correlation_matrix = np.corrcoef(data["TV"], data["Sales"])

correlation_matrix

array([[1.        , 0.78222442],
       [0.78222442, 1.        ]])

In [52]:
# The correlation coefficient between x and y is located at position (0, 1) or (1, 0) in the matrix
correlation_coefficient = correlation_matrix[0, 1]

#Create new column to assign the corelation value
data["TV-Sales"] = correlation_coefficient

print("Pearson's correlation coefficient for the inputs TV-Sales:", correlation_coefficient)

Pearson's correlation coefficient for the inputs TV-Sales: 0.7822244248616061


In [53]:
data

TV  Radio  Newspaper  Sales  Newspaper-Sales  Radio-Sales  TV-Sales
0    230.1   37.8       69.2   22.1         0.228299     0.576223  0.782224
1     44.5   39.3       45.1   10.4         0.228299     0.576223  0.782224
2     17.2   45.9       69.3    9.3         0.228299     0.576223  0.782224
3    151.5   41.3       58.5   18.5         0.228299     0.576223  0.782224
4    180.8   10.8       58.4   12.9         0.228299     0.576223  0.782224
..     ...    ...        ...    ...              ...          ...       ...
195   38.2    3.7       13.8    7.6         0.228299     0.576223  0.782224
196   94.2    4.9        8.1    9.7         0.228299     0.576223  0.782224
197  177.0    9.3        6.4   12.8         0.228299     0.576223  0.782224
198  283.6   42.0       66.2   25.5         0.228299     0.576223  0.782224
199  232.1    8.6        8.7   13.4         0.228299     0.576223  0.782224

[200 rows x 7 columns]

#**Conclusion**

Pearson correlation coeficient, provides important information about the relationship between two variables, it reflets if the relationship is positive or negative and also gives information about the strength of the connection between the variables. In this example the variables TV-Sales are the ones that provides higher correlation with a positive 0.78 value, the combination of the variables Radio-Sales provides a medium correlation with 0.57 and the variables newspaper-Sales, give a low positive correlation with a value of 0.22. In conclusion the variable TV is the one with major impact on the dependent variable, in this scenario the decision makers could take the decision to invest in a marketing strategy using the TV as a main broadcast media.
